flask/app/routes.py

```diff
@@ -356,6 +370,34 @@ def datasets_list():
     return jsonify(datasets)
 
 
+@app.route('/api/datasets/<int:id>', methods=['GET'])
+@login_required
+def get_dataset(id: int):
+    dataset = db.session.query(models.Dataset).filter_by(dataset_id=id).options(
+        joinedload(models.Dataset.analyses),
+        joinedload(models.Dataset.tissue_sample).
+        joinedload(models.TissueSample.participant).
+        joinedload(models.Participant.family)
+     ).one()
+    if not dataset:
```

Comment:
What are the conditions for this? Requesting an ID not found in the table doesn't trigger this as it throws an error on the query (`sqlalchemy.orm.exc.NoResultFound: No row was found for one()`)

Reply:
I was using the wrong method -- should be fixed now.